### PR TITLE
[IMP] blocked_location Adds making the blocked reason required (story…

### PR DESCRIPTION
**Pull request by BlommaertsEdwin**
…/14430)

- Makes the blocked reason required on the interface UI of Odoo
- Executes a migration script setting a default reasons for currently blocked locations with no reason given.

### CODE REVIEW

File: addons/blocked_locations/migrations/0.2/pre-migrate.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+def migrate(cr, version):
+    # Update blocked reasons with dummy reasons since we're making this field a required field if the location is blocked.
+    cr.execute("UPDATE stock_location SET u_blocked_reason = 'Location is blocked' WHERE u_blocked IS True and u_blocked_reason IS Null;")
```
Comment (PeterAlabaster) [inline addons/blocked_locations/migrations/0.2/pre-migrate.py]: nit: format this more like:
```python
    cr.execute("""
        UPDATE stock_location
        SET u_blocked_reason = 'Location is blocked'
        WHERE u_blocked IS True and u_blocked_reason IS Null;
    """) 
```

Comment (RHODOK) [reply addons/blocked_locations/migrations/0.2/pre-migrate.py]: and a new line at the end =) to get rid of the angry red minus sign =D

File: addons/blocked_locations/migrations/0.2/pre-migrate.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+def migrate(cr, version):
+    # Update blocked reasons with dummy reasons since we're making this field a required field if the location is blocked.
```
Comment (williamsajr) [inline addons/blocked_locations/migrations/0.2/pre-migrate.py]: Nit: use """ for docstrings

File: addons/blocked_locations/models/stock_location.py [hunk 3]
```diff
@@ -3,67 +3,74 @@
...
+                    [
+                        ("reserved_quantity", ">", 0),
+                        ("location_id", "=", record.id),
+                    ]
+                )
                 if n_quants > 0:
                     raise ValidationError(
-                            _('Location cannot be blocked because '
-                              'it contains reserved stock.'))
+                        _("Location cannot be blocked because " "it contains reserved stock.")
```
Comment (williamsajr) [inline addons/blocked_locations/models/stock_location.py]: Nit: Don't need two strings here (black formatting doesn't combine strings when removing line breaks)

File: addons/blocked_locations/tests/test_location.py [hunk 4]
```diff
@@ -3,99 +3,116 @@
...
 
         with self.assertRaises(ValidationError) as e:
             self.test_location_01.u_blocked = True
 
-        self.assertEqual(e.exception.name,
-                'Location cannot be blocked because it contains reserved stock.')
+        self.assertEqual(
+            e.exception.name,
+            "A reason for blocking the locations is required when attempting to block a location.",
+        )
```
Comment (williamsajr) [inline addons/blocked_locations/tests/test_location.py:118]: It's probably worth adding a test to explicitly check that when we don't supply a message that an exception is raised

Comment (BlommaertsEdwin) [reply addons/blocked_locations/tests/test_location.py:118]: Hi @williamsajr isn't that what this test already does? :)

Comment (williamsajr) [reply addons/blocked_locations/tests/test_location.py:118]: Yep you're right, I get confused before my first cup of coffee in the morning

File: addons/blocked_locations/models/stock_location.py [hunk 5]
```diff
@@ -3,67 +3,74 @@
...
-            raise ValidationError(
-                    _("Prefix parameter for check_blocked should be string"))
+            raise ValidationError(_("Prefix parameter for check_blocked should be string"))
         blocked_locations = self.filtered(lambda x: x.u_blocked)
         if blocked_locations:
             raise ValidationError(
-                    _('%s %s Please speak to a team leader '
-                      'to resolve the issue.') %
-                      (prefix, ' '.join(blocked_locations._prepare_blocked_msg())))
+                _("%s %s Please speak to a team leader " "to resolve the issue.")
```
Comment (williamsajr) [inline addons/blocked_locations/models/stock_location.py]: Another double string here

File: addons/blocked_locations/migrations/0.2/pre-migrate.py (hunk 2 above)
Comment (williamsajr) [reply addons/blocked_locations/migrations/0.2/pre-migrate.py]: Nit for fix: Line length is now 128 characters

**Review by williamsajr — APPROVED** [review]

**Comment by adamrpatrick** [comment]
Bit late now for this one, but can we keep formatting changes in a separate commit please? It makes code reviews much easier! Thanks

**Pull request merged**